src/routes/decks/[marketplaceDeckId]/+page.svelte

```diff
@@ -0,0 +1,98 @@
+<script lang="ts">
+  import { user, willAttemptLogin } from "$lib/firebase";
+  import { enhance } from "$app/forms";
+  import type { PageData } from "./$types";
+  import { afterUpdate } from "svelte";
+  import ProBadgeWrapper from "$lib/components/ProBadgeWrapper.svelte";
+  import { goto } from "$app/navigation";
+  export let data: PageData;
+
+  $: {
+    if (!$user && !willAttemptLogin())
+      goto(`/login?successRedirect=${encodeURIComponent(location.pathname)}`);
+  }
+
+  let isPro = false;
+  let isLoading = true;
+
+  afterUpdate(async () => {
```

Comment:
Why did you change this from `onMount`? This has the potential to run many times